File: app/models/language.rb
# == Schema Information
#
# Table name: languages
#
#  id           :bigint           not null, primary key
#  long_name    :string           default("english"), not null
#  long_region  :string
#  short_name   :string           default("en"), not null
#  short_region :string
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#
class Language < ApplicationRecord
  has_many :users
  has_many :tests
end

File: dotnet/ManagementHub.Storage/Data/Language.cs
using System;
using System.Collections.Generic;
using ManagementHub.Models.Abstraction;

namespace ManagementHub.Models.Data;

public partial class Language : IIdentifiable
{
	public Language()
	{
		this.Tests = new HashSet<Test>();
		this.Users = new HashSet<User>();
	}

	public long Id { get; set; }
	public string LongName { get; set; } = null!;
	public string ShortName { get; set; } = null!;
	public string? LongRegion { get; set; }
	public string? ShortRegion { get; set; }
	public DateTime CreatedAt { get; set; }
	public DateTime UpdatedAt { get; set; }

	public virtual ICollection<Test> Tests { get; set; }
	public virtual ICollection<User> Users { get; set; }
}
